diff --git modules/linkit/src/Plugin/Linkit/Substitution/Media.php modules/linkit/src/Plugin/Linkit/Substitution/Media.php
index 776a0ae..b630961 100644
--- modules/linkit/src/Plugin/Linkit/Substitution/Media.php
+++ modules/linkit/src/Plugin/Linkit/Substitution/Media.php
@@ -42,6 +42,15 @@ class Media extends PluginBase implements SubstitutionInterface, ContainerFactor
     }

     $source_field = $entity->getSource()->getSourceFieldDefinition($entity->get('bundle')->entity);
+
+    // patch for CWRU HR documents
+    if ($entity->get('bundle')->getValue()[0]['target_id'] == 'hr_documents') {
+      $fid = $entity->get('field_hr_form_document')->getValue()[0]['target_id'];
+      $file_entity = \Drupal\file\Entity\File::load($fid);
+      $file_url_generator = \Drupal::service('file_url_generator');
+      return $file_url_generator->generate($file_entity->getFileUri());
+    }
+
     if ($source_field && $entity->hasField($source_field->getName()) && $entity->get($source_field->getName())->entity instanceof FileInterface) {
       /** @var \Drupal\file\FileInterface $file */
       $file = $entity->get($source_field->getName())->entity;
